internal/module/currency/api/handler/register.go

```diff
@@ -0,0 +1,13 @@
+package handler
+
+import "github.com/gin-gonic/gin"
+
+type Interface interface {
+	GetLatest(ctx *gin.Context)
+}
```

Comment:
This interface makes no sense, since its method signature depends on the details of the infrastructure (gin). You can use explicit dependencies because this is the infrastructure layer and there is no reason to use DIP here.